datafusion/core/src/physical_plan/stream.rs

```diff
@@ -169,7 +171,16 @@ impl RecordBatchReceiverStreamBuilder {
         let check = async move {
             while let Some(result) = join_set.join_next().await {
                 match result {
-                    Ok(()) => continue, // nothing to report
+                    Ok(task_result) => {
+                        match task_result {
+                            // nothing to report
+                            Ok(_) => continue,
+                            // This means a blocking task error
+                            Err(e) => {
+                                return Some(internal_err!("Spawned Task error: {e}"));
```

Comment:
For example, after this, if the file doesn't exist, you will get an error:

```
Error: Internal("Spawned Task error: Execution error: Spill file \"/var/folders/zq/2tdnn5955wvdcw7qfy6qhvk00000gn/T/.tmpb5EGzS/.tmptpkruZ\" does not exist")
```

Reply:
`ExecutionError` might be more appropriate than `InternalError` given an error in a spill file is likely not a bug in DataFusion, but rather some issue in the environment.